pkg/client/status_reporter.go

```diff
@@ -77,7 +77,7 @@ func (r *StatusReporter) SetDone() error {
 	conditions := newConditions(co.Status, r.version, time)
 	conditions.setCondition(v1.OperatorAvailable, v1.ConditionTrue, "Successfully rolled out the stack.", "RollOutDone", time)
 	conditions.setCondition(v1.OperatorProgressing, v1.ConditionFalse, "", "", time)
-	conditions.setCondition(v1.OperatorDegraded, v1.ConditionFalse, "", "", time)
+	conditions.setCondition(v1.OperatorDegraded, v1.ConditionFalse, degradedConditionMessage, "", time)
```

Comment:
We probably should set the actual condition here depending on the message passed?

Reply:
Also this should add a reason. Maybe `PrometheusNoDataPersistence` or something like it

Reply:
> We probably should set the actual condition here depending on the message passed?

Do you mean we should set the condition to `true` when a message other than `""` is passed?
I probably should have provided context in the PR description, the requirement is to actually keep the condition to `degraded=false` but set a message for cluster admins to be aware of the state. I update the PR description to contain that information.

Reply:
> Also this should add a reason. Maybe PrometheusNoDataPersistence or something like it

Good point, added the reason as well.

Reply:
> Do you mean we should set the condition to `true` when a message other than `""` is passed?

Yes that is the convention afaiu, at least in the other direction (don't set the condition without also setting a reason). To me it makes sense too, but I'll try to find some according docs.

Reply:
@wking we could not find any documentation on this particular topic. Is it okay for an operator to set a message for the `degraded` condition but keep the value to `false`. This is what we discussed for CMO when there is no persistent storage enabled to indicate to the admin that metrics might be lost.

Reply:
Based on Trevor's reply in slack, setting a message for the degraded condition while keeping the value to false is a valid state for the operator status. A good practice in these situations is to add a link to the documentation on how to fix the underlying problem.